### DISCUSSION TITLE
Trying to figure out media-lab script for imgur.com

### DISCUSSION THREAD

**Discussion by The-Sound** [uBlockOrigin/uBlock-issues#2594] [Q&A]
The "new beta"  layout on imgur has been having js.media-lab.ai as a cross script since the adquisition.

I was playing around to fix something broken on the web when I whitelisted it.
The thing is, after whitelisting it, I can't go back to blacklisting it because it will refuse to load properly.

I've tried clearing cookies, local stoarge, local DB etc etc, but seems like once I let the script in, I got "marked", even knowing that makes no sense, is the closest comparisson I can think of.

What I would really love is for me to debug it properly, since I use some tweaks on firefox, although I can replicate the same behavior on a default chrome installation.

![image](https://user-images.githubusercontent.com/24360204/231246444-eeb50234-ef9d-4cab-8eb0-7732fffbac7d.png)

**Comment by gorhill** [comment]
Red badge means you are blocking all 3rd parties. I can make it works blocking all 3rd parties by noop-ing `js.media-lab.ai` and `imgur.map.fastly.net`.

**Reply by The-Sound** [reply]
Yes, that's my experience as well. And the intended usage I've been having.

The mystery for me is, that I do not want to noop media-lab, and I _wasn't_ nooping it before, and it was working.
I nooped it to test some things, and when I reverted, it won't load, like in the image added.
So if I want to use imgur, now I'm _forced_ to noop media-lab, which wasn't the case before.

I hope I'm making sense.

**Reply by gorhill** [reply]
What browser is this? It could be that the required script from `js.media-lab.ai` is in the memory cache, in which case this would bypass uBO's `webRequest` listener.

**Reply by The-Sound** [reply]
It is firefox 111.0.1 (64-bit) on linux.
If you mean desktop's memory, I also drop ram's cache and delete mozilla's cache folder after cleaning all data related to imgur and closing the browser.

**Reply by gorhill** [reply]
Well in the end you have to use the logger to find out what uBO is doing internally, so it is possible a dynamic URL rule was present, I won't be able to guess more than I do why it worked on your side.

**Reply by The-Sound** [reply]
My belief is, that I think that something done _before_ uBo can do its thing, but I lack the knowledge or tools to debug properly, and that's why I came here to see if I could get some help.

I don't see anything weird in the logger, it just loads up the gray div you see in the initial image.
I've tested whitelisting some blocked things from some filters, just in case, but did no difference.
The only constante behavior is:
- I need to yes or yes, whitelist js.media-lab.ai
- Once you whitelist it, you can't blacklist back or website will break.

And that's why I'm mostly curious to know how does it work and why does it work that way.

**Comment by mapx-** [comment]
Did you examine "my rules" and "trusted sites" sections ?
If you see there strange rules (related to imgur)  just remove them and test again.

or .. reset uBO / reinstall it

**Reply by The-Sound** [reply]
Ok, I've been roaming the logger around again since I had nothing special in my rules, and I believe this _could_ be the culprit. Again, I have almost zero knowledge on the topic, so correct me as needed.

Seems like once whitelisted, `js.media-lab.ai` will load 2 scripts:
```
https://js.media-lab.ai/wana.2.4.1.js
https://js.media-lab.ai/ana-sentry.js
```

When whitelisted, EasyPrivacy will block `ana-sentry` but it will load up `wana`
Delving a little more into the code, by clicking on the logger link, I've found the next piece of code, which again, is just a random guess from me:
``` js
    d = "ana_client_uid",
    c = "ana_client_session_id",
    l = "ana_vast",
    u = "ana_staging",
```
I believe this is why once I whitelisted it, it created or asigned an ID to me, and then it tries to do some checks, that I block through uBO.

There's also a line that would set the background colour but this could be just whatever, adding it here just in case:
```js
function M() {
      for (var e = [], t = 0; t < arguments.length; t++) e[t] = arguments[t];
      P() && console.log.apply(console, D(["%c🚨ANA: ", "background: #222; color: #33c65a"], e, !1))
   }
